### PR TITLE
Human-readable EDTF display transformation

### PR DESCRIPTION
**Pull request by rschenk**
## Description

Now that dates are stored as edtf strings in the database, we can display them in human readable form in the UI. For example "1984?" becomes "Circa 1984." The actual humanizing functionality comes from `duke-libraries/edtf-humanize`.

Connected to #687 and #1035 

## Changes

* Display transformation for EDTF dates
* Data Dictionary changes for edtf dates:
  * change storage from date type to string type
  * remove date index (superseded by #726)
  * add humanize edtf display transformation
* Remove `Indexing::Dates` as it is no longer used (superseded by #726)

### CODE REVIEW

**Comment by rschenk** [comment]
Coveralls is having a gripper, but honestly I can't tell why it's doing so. None of the files where coverage dropped were anything related to what I changed

**Comment by awead** [comment]
@rschenk bollocks to coveralls I say!

File: spec/cho/work/submissions/show_spec.rb [hunk 1]
```diff
@@ -22,6 +23,8 @@
         expect(page).to have_blacklight_field('work_type_ssim').with('Document')
         expect(page).to have_blacklight_label('home_collection_id_tesim')
         expect(page).to have_blacklight_field('home_collection_id_tesim').with('Library Collection')
+        expect(page).to have_blacklight_label('created_tesim')
+        expect(page).to have_blacklight_field('created_tesim').with('Circa 1981')
```
Comment (awead) [inline spec/cho/work/submissions/show_spec.rb]: nice!!!!

Comment (rschenk) [reply spec/cho/work/submissions/show_spec.rb]: 😎

**Pull request merged**